integr8ly/integreatly-operator
# Multitenant metrics

# Issue link
[MGDAPI-2648](https://issues.redhat.com/browse/MGDAPI-2648) 
[MGDAPI-2662](https://issues.redhat.com/browse/MGDAPI-2662) 

# What
Metrics for Multi Tenancy
Scripts for Multi Tenancy

# Verification steps
Build an image based on this branch
`ORG=<yourrepo> make image/build/push`

Install multi tenant RHOAM Operator 
`INSTALLATION_TYPE=multitenant-managed-api make install/sandboxrhoam/operator`
After the Operator Installs change the CSV to reference your new image above 
Then create config including, rhmi cr
`INSTALLATION_TYPE=multitenant-managed-api make install/sandboxrhoam/config`

After RHSSO cluster installs, install the IDP
`NUM_REGULAR_USER=10 NUM_ADMIN=5 REALM_DISPLAY_NAME="DevSandbox" REALM="devsandbox" DEDICATED_ADMIN_PASSWORD=Password1 PASSWORD=Password1 ./scripts/setup-sso-idp.sh`

Go to the terminal of the rhoam operator and run the following to monitor the new metrics:
`while sleep 1; do curl -s localhost:8383/metrics | grep no_; done`

This will monitor for 2 out of the three new metrics `no_tenant_realm` and `no_activated_3scale_tenant_account`
As tenants are created you should see these metrics being populated and subsequently being depleted. 
Login to OpenShift as test-user01, test-user02, test-user03, test-user04, test-user05 to create tenants. 
Verify that the above metrics get populated with the usernames temporarily before being remove completely.

Also verify:
`curl -s localhost:8383/metrics | grep num_tenants`   == 5
